packages/framework/src/Model/Order/PromoCode/CurrentPromoCodeFacade.php

```diff
@@ -97,14 +98,17 @@ protected function validatePromoCodeByProductsInCart(PromoCode $promoCode, Cart
         $allowedProductIds = array_unique(array_merge($allowedProductIds, $allowedProductIdsByCriteria));
 
         if (count($allowedProductIds) === 0) {
-            //promo code hasn't any relation with products or product from categories or product from brands
-            return;
+            // promo code hasn't any relation with products or product from categories or product from brands
+            return array_map(
+                fn (Product $product) => $product->getId(),
+                $products,
+            );
```

Comment:
I do not understand this part :smile:

Reply:
basically "return all product ids from passed products"